### DISCUSSION TITLE
I am not able to mint nft for lesson 3

### DISCUSSION THREAD

**Discussion by Shiv2116** [Cyfrin/foundry-full-course-cu#110] [Q&A] [answered]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/114937365/24ecc497-38bd-4211-bf5b-4012f3a26faf)

**Comment by alymurtazamemon** [comment]
@Shiv2116 Make sure these 2 points;

1. You cannot mint twice with the same address.
2. If you did not mint it before with this address and seeing this gas issue it means you have not solved the challenge yet.

**Reply by ppoojjaa** [reply]
where do I find the challenge?

**Reply by Shiv2116** [reply]
at the end of every lesson
there are two qr codes to be scanned
One for testnet and one for mainnet

**Reply by ppoojjaa** [reply]
yes, I got that but for example in lesson 2, you have to input a number and twitter handle. So how do I find the correct number? Where is the challenge to find that number?

**Comment by Shiv2116** [comment]
for the first point concerned, i havent made any txn yet and still in metamask it shows " not able to estimate the gas price"
what if I click still proceed anyway? will it solve the issue?

**Reply by alymurtazamemon** [reply]
@Shiv2116 No

**Comment by Shiv2116** [comment]
what has to be done
my answers are correct

**Reply by alymurtazamemon** [reply]
@Shiv2116 Your answer is not correct. You can come back to these challenges after finishing the course.

**Comment by Shiv2116** [comment]
the true boolean is in index 3 right?

**Comment by PatrickAlphaC** [comment]
hehehehehehehhe

**Comment by npa528** [comment, marked as answer]
@Shiv2116 
you don't always need to use "Write Contract" **only**.

You should read the contract to find the real locations of boolean variables. Users that solve this Challenge move this variable to different index, so you have to read the latest index location and use it accordingly.

**Reply by ShubhSensei** [reply]
Thanx for the help buddy. 😁👍

**Reply by 0xtdey** [reply]
this is the best hint

**Comment by matthewg49** [comment]
@Shiv2116 had the same issue and just wanted to say thank you for making this post! 

For anyone still stick on this challenge, there's enough hints in this thread that will yield the results ;)

**Comment by 0xArDANT** [comment]
This post is so helpful, I just solved the challenge.
 Thank you guys, you're the best.

**Comment by EngrPips** [comment]
Interesting
